### PR TITLE
feat: add-logging

### PR DESCRIPTION
**Pull request by JGSweets**
in this pr:
 * add logging
 * fixes bug with attorney and attorney org assignment in dict
 * allows for scanning folders for files
 * uses better org and name building

### CODE REVIEW

File: setup_loggers.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import os
```
Comment (JGSweets) [inline setup_loggers.py:1]: logger file

File: setup_loggers.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import sys
+import logging
+
+
+logFormatter = logging.Formatter("%(asctime)s [%(threadName)-12.12s] [%(levelname)-5.5s]  %(message)s")
+
+
+def setup_root_logger(level: int = logging.INFO):
```
Comment (JGSweets) [inline setup_loggers.py:10]: sets to console for root

File: setup_loggers.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    root_logger = logging.getLogger()
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setFormatter(logFormatter)
+    root_logger.setLevel(level)
+    root_logger.addHandler(handler)
+
+
+def setup_file_logger(filename: str, level: int = logging.INFO) -> logging.Logger:
```
Comment (JGSweets) [inline setup_loggers.py:19]: sets up logging to a file

File: parse_patent.py [hunk 4]
```diff
@@ -343,26 +367,54 @@ def write_to_db(uspto_patent, db=None):
...
+
+if __name__ == "__main__":
+    _arg_filenames = []
+    if len(sys.argv) > 1:
+        _arg_filenames = sys.argv[1:]
+
 
-print("\n\nErrors\n------------------------\n")
-for e in errors:
-    print(e)
```
Comment (JGSweets) [inline parse_patent.py:365]: moved and uses logger

File: parse_patent.py [hunk 5]
```diff
@@ -343,26 +367,54 @@ def write_to_db(uspto_patent, db=None):
...
+    _db_config_file = "config/postgres.tsv"
+    _db = PGDBInterface(config_file=_db_config_file)
+    _db.silent_logging = True
 
-print("Success Count:", success_count)
-print("Error Count:", len(errors))
```
Comment (JGSweets) [inline parse_patent.py:368]: moved and uses logger

File: parse_patent.py [hunk 6]
```diff
@@ -343,26 +367,54 @@ def write_to_db(uspto_patent, db=None):
...
+    _push_to = "db"
+    load_local_files(
+        dirpath_list=_arg_filenames,
+        limit_per_file=None,
```
Comment (JGSweets) [inline parse_patent.py:417]: can set a limit of how many patents to load per file

File: parse_patent.py [hunk 7]
```diff
@@ -343,26 +367,54 @@ def write_to_db(uspto_patent, db=None):
...
-                db.commit_to_db()
+                logger.info(f"{count}, {filename}, {title}")
+                if push_to.endswith(".jsonl"):
+                    with open(push_to, "a") as fp:
+                        fp.writelines(patent_dumps_list)
+                    patent_dumps_list = []
+                _db.commit_to_db()
             count += 1
 
+    if errors:
```
Comment (JGSweets) [inline parse_patent.py:396]: was below

**Review by lettergram — APPROVED** [review]

**Pull request merged**